pkg/marker/marker.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	clientset kubernetes.Interface
+)
+
+const (
+	// Expose available bridges as resources in format bridge.network.kubevirt.io/[bridge name]
+	resourceNamespace = "bridge.network.kubevirt.io"
+	// Kubernetes API does not support infinite resources, assume that 1000 connections is enough
+	resourceDefaultValue = "1000"
+)
+
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+func init() {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf("Error while obtaining cluster config: %v", err)
+	}
+
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Error building example clientset: %v", err)
+	}
+}
+
+func getAvailableResources() (map[string]bool, error) {
```

Comment:
can we perhaps use netlink go bindings instead of parsing a command output? seems rather brittle.

Reply:
github.com/vishvananda/netlink?
What node resources would it need?

Alternatively we could check in /sys/class/net

Reply:
Please clarify what you mean by "node resources".

The idea is to not rely on output of a tool that is prone to break if output is changed in the future; but to use proper bindings to interact with the kernel.

Reply:
Please use vishavananda netlink lib. I think it is the one most often used, even in KubeVirt code.

Reply:
Changed to check /sys/class/net. IMHO this looks even better.
What dost thou think?

Reply:
I prefer we don't parse `/sys/class/net` ourselves but maybe it's just me. It's definitely better like that than parsing `ip` output which is a lot more prone to breakage.

Reply:
Switched to netlink